src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -99,17 +99,53 @@ private void validExistedStation(boolean isExistedUpStation, boolean isExistedDo
     }
 
     private void isValidDeleteSection(Long stationId) {
-        if (getLastSection().getDownStation().getId() != stationId) {
-            throw new IllegalArgumentException("종점역만 삭제가 가능합니다.");
-        }
         if (size() == 1)
             throw new RuntimeException("마지막 구간은 삭제할 수 없습니다.");
     }
 
+    private List<Station> getUpStationList(){
```

Comment:
자료구조에 종속적인 메서드 네이밍이 될 수 있을 것 같아요!

```suggestion
    private List<Station> getUpStations(){
```